plugin/formatting.py

```diff
@@ -50,9 +50,11 @@ def format_document(text_command: LspTextCommand, formatter: Optional[str] = Non
     return Promise.resolve(None)
 
 
-def apply_text_edits_to_view(response: Optional[List[TextEdit]], view: sublime.View) -> None:
+def apply_text_edits_to_view(
+    response: Optional[List[TextEdit]], view: sublime.View, *, process_placeholders: bool = False
+) -> None:
     edits = list(parse_text_edit(change) for change in response) if response else []
-    view.run_command('lsp_apply_document_edit', {'changes': edits})
+    view.run_command('lsp_apply_document_edit', {'changes': edits, 'process_placeholders': process_placeholders})
```

Comment:
It's a bit confusing that the `process_placeholders` is also added to this function, even though it is not used anywhere in the code. I see that you used it in the rust-analyzer PR, but `apply_text_edits_to_view` is not part of the public API...

I think it would be better if LspApplyDocumentEditCommand would simply take a list of `TextEdit | AnnotatedTextEdit` and optional `version` argument, instead of a list of custom `TextEditTuple`. With the current design there is no way to properly use LspApplyDocumentEditCommand from the outside when you have no access to `parse_text_edit`. I wonder why this part, which is only a single line (line 56 here), is not simply part of the command anyway to reduce code duplication. Here are the relevant usages of the command, all with the same additional code required before:
```
C:\Users\jwortmann\AppData\Roaming\Sublime Text\Packages\LSP\plugin\color.py:
   59              color_pres = self._filtered_response[index]
   60              text_edit = color_pres.get('textEdit') or {'range': self._range, 'newText': color_pres['label']}
   61:             self.view.run_command('lsp_apply_document_edit', {'changes': [parse_text_edit(text_edit, self._version)]})
   62  

C:\Users\jwortmann\AppData\Roaming\Sublime Text\Packages\LSP\plugin\completion.py:
  373          if additional_edits:
  374              edits = [parse_text_edit(additional_edit) for additional_edit in additional_edits]
  375:             self.view.run_command("lsp_apply_document_edit", {'changes': edits})
  376          command = item.get("command")
  377          if command:

C:\Users\jwortmann\AppData\Roaming\Sublime Text\Packages\LSP\plugin\formatting.py:
   53  def apply_text_edits_to_view(response: Optional[List[TextEdit]], view: sublime.View) -> None:
   54      edits = list(parse_text_edit(change) for change in response) if response else []
   55:     view.run_command('lsp_apply_document_edit', {'changes': edits})
```

Reply:
I'm not sure why is that. I imagine potential reasons could be to optimize the payload size or avoid passing unserializable data but neither of those is relevant here IMO.

As for public vs. non-public, we both know that there are many "non-public" APIs that are used from plugins. People will use whatever they need since we don't prevent them :)